compute/baremetal.py

```diff
@@ -0,0 +1,139 @@
+"""Support VM lifecycle operation in an OpenStack Cloud."""
+import logging
+from typing import List, Optional
+
+from libcloud.compute.base import Node
+
+LOG = logging.getLogger()
+
+# libcloud does not have a timeout enabled for Openstack calls to
+# ``create_node``, and it uses the default timeout value from socket which is
+# ``None`` (meaning: it will wait forever). This setting will set the default
+# to a magical number, which is 280 (4 minutes). This is 1 minute less than the
+# timeouts for production settings that should allow enough time to handle the
+# exception and return a response
+
+
+# Custom exception objects
+class ResourceNotFound(Exception):
+    pass
+
+
+class ExactMatchFailed(Exception):
+    pass
+
+
+class VolumeOpFailure(Exception):
+    pass
+
+
+class NetworkOpFailure(Exception):
+    pass
+
+
+class NodeError(Exception):
+    pass
+
+
+class NodeDeleteFailure(Exception):
+    pass
+
+
+class CephBaremetalNode:
+    """Represent the VMNode required for cephci."""
+
+    def __init__(
+        self,
+        **params,
+    ) -> None:
+        """
+        Initialize the instance using the provided information.
+
+        The co
+
+        Args:
+            username:   The name of the user to be set for the session.
+            password:   The password of the provided user.
+            auth_url:   The endpoint that can authenticate the user.
+            auth_version:   The version to be used for authentication.
+            tenant_name:    The name of the user's project.
+            tenant_domain_id:   The ID of the user's project.
+            service_region: The realm to be used.
+            domain_name:    The authentication domain to be used.
```

Comment:
Please fix the docstring... It is helpful, if you could document the required arguments along with the optional ones.